tests/aggregation/tests.py

```diff
@@ -2158,6 +2159,45 @@ def test_having_with_no_group_by(self):
         )
         self.assertEqual(list(author_qs), [337])
 
+    def test_JSONArrayAgg(self):
+        vals = Store.objects.aggregate(jsonarrayagg=JSONArrayAgg("name"))
+        self.assertEqual(
+            vals,
+            {"jsonarrayagg": ["Amazon.com", "Books.com", "Mamma and Pappa's Books"]},
+        )
+
+    def test_JSONArrayAgg_datefield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pubdate"))
+        self.assertEqual(
+            vals,
+            {
+                "jsonarrayagg": [
+                    "2007-12-06",
+                    "2008-03-03",
+                    "2008-06-23",
+                    "2008-11-03",
+                    "1995-01-15",
+                    "1991-10-15",
+                ]
+            },
+        )
+
+    def test_JSONArrayAgg_decimalfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("price"))
+        self.assertEqual(
+            vals, {"jsonarrayagg": [30.0, 23.09, 29.69, 29.69, 82.8, 75.0]}
+        )
+
+    def test_JSONArrayAgg_integerfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pages"))
+        self.assertEqual(vals, {"jsonarrayagg": [447, 528, 300, 350, 1132, 946]})
+
+    def test_JSONArrayAgg_filter(self):
+        vals = Author.objects.filter(age__gt=29).aggregate(
+            jsonarrayagg=JSONArrayAgg("age")
```

Comment:
I don't think we need to test `filter.aggregate` as it has little to do with `JSONArrayAgg` itself.

We should cover the [different properties](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#aggregate-filter) that it inherits from `Aggregate` instead such as 

1. `distinct=True`
2. `filter=Q(...)`
3. `default`

Another area that should be covered are null handling.